components/src/core/ToolbarMenu/ToolbarMenu.tsx

```diff
@@ -0,0 +1,258 @@
+            }
+            setAnchorEl(event);
+        },
+        [onOpen]
+    );
+
+    useEffect(() => {
+        if (menuGroups) {
+            for (const group of menuGroups) {
+                for (const item of group.items) {
+                    const onClick = item.onClick;
+                    if (onClick) {
+                        item.onClick = (e: React.MouseEvent<HTMLElement, MouseEvent>): void => {
+                            onClick(e);
+                            closeMenu();
+                        };
+                    }
+                }
+            }
+        }
+    }, [menuGroups]);
+
+    const getMenu = useCallback(() => {
+        if (menu && Boolean(anchorEl)) {
+            return React.cloneElement(menu, {
+                anchorEl: anchorEl,
+                open: Boolean(anchorEl),
+                onClose: closeMenu,
+                ...menu.props,
+            });
+        }
+        if (menuGroups && Boolean(anchorEl)) {
+            return (
+                <Menu
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: rtl ? 'right' : 'left' }}
+                    transformOrigin={{ vertical: 'top', horizontal: rtl ? 'right' : 'left' }}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => closeMenu()}
+                    {...MenuProps}
+                    MenuListProps={{ style: { padding: 0 } }}
+                >
+                    {!menu &&
+                        menuGroups.map((group: ToolbarMenuCompGroup, index: number) => (
+                            <div className={defaultClasses.navGroups} key={index}>
+                                <DrawerNavGroup
+                                    divider={false}
+                                    hidePadding={true}
+                                    itemIconColor={group.iconColor}
+                                    itemFontColor={group.fontColor}
+                                    title={group.title}
+                                    items={group.items.map(
+                                        (item: ToolbarMenuCompItem, itemIndex: number): NavItem =>
+                                            Object.assign({ itemID: itemIndex.toString() }, item)
+                                    )}
+                                />
+                            </div>
+                        ))}
+                </Menu>
+            );
+        }
+    }, [menuGroups, menu, anchorEl, MenuProps, defaultClasses]);
+
+    return (
+        <>
+            <Typography
+                ref={composeRefs(ref, anchor)}
+                aria-haspopup="true"
+                {...otherTypographyProps}
+                className={clsx(
+                    defaultClasses.root,
+                    classes.root,
+                    menuGroups || menu ? defaultClasses.cursorPointer : ''
+                )}
```

Comment:
Don't forget the className prop that can be passed in as well (otherwise we are overwriting it instead of merging)
```suggestion
                className={clsx(
                    defaultClasses.root,
                    classes.root,
                    props.className,
                    menuGroups || menu ? defaultClasses.cursorPointer : ''
                )}
```